File: app/src/main/java/workshop1024/com/xproject/model/publisher/Publisher.java
package workshop1024.com.xproject.model.publisher;


import android.databinding.ObservableBoolean;

/**
 * 发布者数据类
 */
public class Publisher {
    //发布者id
    private String mPublisherId;
    //发布者类型
    private String mTypeId;
    //发布者语言类型
    private String mLanguage;
    //发布者图标URL
    private String mIconUrl;
    //发布者名称
    private String mName;
    //发布者订阅数量
    private String mSubscribeNum;
    //发布者是否被订阅
    public final ObservableBoolean isSubscribed = new ObservableBoolean();

    public Publisher(String publisherId, String type, String language, String iconUrl, String name,
                     String subscribeNum, boolean isSubscribed) {
        mPublisherId = publisherId;
        mTypeId = type;
        mLanguage = language;
        mIconUrl = iconUrl;
        mName = name;
        mSubscribeNum = subscribeNum;
        this.isSubscribed.set(isSubscribed);
    }

    public String getPublisherId() {
        return mPublisherId;
    }

    public void setPublisherId(String publisherId) {
        mPublisherId = publisherId;
    }

    public String getTypeId() {
        return mTypeId;
    }

    public void setTypeId(String typeId) {
        mTypeId = typeId;
    }

    public String getLanguage() {
        return mLanguage;
    }

    public void setLanguage(String language) {
        mLanguage = language;
    }

    public String getIconUrl() {
        return mIconUrl;
    }

    public void setIconUrl(String iconUrl) {
        mIconUrl = iconUrl;
    }

    public String getName() {
        return mName;
    }

    public void setName(String name) {
        mName = name;
    }

    public String getSubscribeNum() {
        return mSubscribeNum;
    }

    public void setSubscribeNum(String subscribeNum) {
        mSubscribeNum = subscribeNum;
    }

    @Override
    public String toString() {
        return "Publisher{" +
                "mPublisherId='" + mPublisherId + '\'' +
                ", mTypeId='" + mTypeId + '\'' +
                ", mLanguage='" + mLanguage + '\'' +
                ", mIconUrl='" + mIconUrl + '\'' +
                ", mName='" + mName + '\'' +
                ", mSubscribeNum='" + mSubscribeNum + '\'' +
                ", mIsSubscribed=" + isSubscribed +
                '}';
    }
}

File: main/src/main/java/workshop1024/com/xproject/main/model/publisher/Publisher.kt
package workshop1024.com.xproject.main.model.publisher

import androidx.databinding.ObservableBoolean
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * 发布者数据类
 */
@Entity(tableName = "publishers")
data class Publisher( //发布者id
        @PrimaryKey @ColumnInfo(name = "publisherId") var mPublisherId: String, //发布者
        @ColumnInfo(name = "mType") var mType: String,//发布者语言类型
        @ColumnInfo(name = "language") var mLanguage: String,//发布者图标URL
        @ColumnInfo(name = "iconUrl") var mIconUrl: String,//发布者名称
        @ColumnInfo(name = "mName") var mName: String,//发布者订阅数量
        @ColumnInfo(name = "subscribeNum") var mSubscribeNum: String,//发布者是否被订阅
        @ColumnInfo(name = "isSubscribed") var mIsSubscribed: ObservableBoolean //使用观察者数据对象自动刷新，参考：https://developer.android.com/topic/libraries/data-binding/observability
)